pulp_ansible/app/tasks/collections.py

```diff
@@ -131,10 +131,34 @@ def import_collection(artifact_pk):
             collection, created = Collection.objects.get_or_create(
                 namespace=collection_info["namespace"], name=collection_info["name"]
             )
-            collection_version = CollectionVersion(
-                collection=collection, version=collection_info["version"]
-            )
+            if created:
+                CreatedResource.objects.create(content_object=collection)
+
+            dependencies = collection_info.pop('dependencies')
+            tags = collection_info.pop('tags')
+
+            # Mazer returns many None values. We need to let the default='' in models.py prevail
+            for key in ['description', 'documentation', 'homepage', 'issues', 'license_file',
+                        'readme', 'repository']:
+                if collection_info[key] is None:
+                    collection_info.pop(key)
+
+            collection_version = CollectionVersion(collection=collection, **collection_info)
             collection_version.save()
+
+            for dependency in dependencies.values():
+                dependency_split = dependency.split('.')
+                namespace = dependency_split[0]
+                name = dependency_split[1]
+                found_dependency = CollectionVersion.objects.get(namespace=namespace, name=name)
+                collection_version.dependencies.add(found_dependency)
```

Comment:
We want to fail import if a dependency + matching version is not found (FYI, can be in a later PR). Note: while community galaxy checks for these dependencies, it doesn’t store them as a FK (but that change may have already be discussed/decided). Galaxy’s dep check: https://github.com/ansible/galaxy/blob/028a0ae075d5497ef43b6704c431a6f4c19b4162/galaxy/worker/importers/collection.py#L28

Reply:
Yes this will cause that behaviour I expect. At L#153 the get() fetching 0 items will fail.

Reply:
It'll also cause a ` MultipleObjectsReturned` exception if fetching `>1` items.